src/ui/components/Location/component.ts

```diff
@@ -124,78 +108,104 @@ export default class LocationComponent extends Component {
+  sortMeasurements(left, right) {
+    if (left > right) {
+      return 1;
+    } else if (left < right) {
+      return -1;
+    }
+    return 0;
+  }
 
-    let readFromCache = () => {
-      this.location = store.cache.query(locationQuery);
-      this.measurements = store.cache.query(measurementQuery);
-    };
+  readFromCache(locationSignature: RecordSignature) {
+    let { store } = this.args;
+    this.location = store.cache.query((q) => q.findRecord(locationSignature));
+    this.measurements = store.cache.query((q) =>
+      q.findRelatedRecords(locationSignature, 'measurements')
+    );
+  }
 
+  async loadFromInlineCache(locationSignature: RecordSignature) {
     try {
-      // always try loading data from cache
-      readFromCache();
+      let { isSSR, localStore } = this.args;
+
+      this.readFromCache(locationSignature);
 
       // work around a bug in Orbit.js - see https://github.com/orbitjs/orbit/issues/476
       if (this.recordsFound && !isSSR) {
-        await localStore.push((t) => t.replaceRelatedRecords(
-          locationSignature,
-          'measurements',
-          this.measurements
-        ));
+        await localStore.push((t) =>
+          t.replaceRelatedRecords(
+            locationSignature,
+            'measurements',
+            this.measurements
+          )
+        );
       }
     } catch {
-      // fall through to other loading options…
+      // Allow other methods to find data
     }
+  }
 
-    if (!isSSR) {
-      try {
-        await pullIndexedDB();
-        // try loading data from cache again after IndexedDB has been restored
-        readFromCache();
-      } catch {
-        // fall through to other loading options…
+  async loadFromIndexedDB(locationSignature: RecordSignature) {
+    try {
+      await this.args.pullIndexedDB();
+      // try loading data from cache again after IndexedDB has been restored
+      this.readFromCache(locationSignature);
+    } catch {
+      // Allow other methods to find data
+    }
+  }
+
+  async loadFromAPI(locationSignature: RecordSignature) {
+    let { store } = this.args;
+    try {
+      let location: Location = await store.query((q) =>
+        q.findRecord(locationSignature)
+      );
+      let measurements: Measurement[] = await store.query((q) =>
+        q.findRelatedRecords(locationSignature, 'measurements')
+      );
+
+      if (location && measurements) {
+        this.location = location;
+        this.measurements = measurements;
+
+        // Remember we saw this location
+        let currentDate = new Date().toISOString();
+        store.update((t) =>
+          t.replaceAttribute(locationSignature, 'visitedAt', currentDate)
+        );
       }
 
-      if (!this.recordsFound) {
-        this.loading = true;
+      // If records were found, update fog effect
+      if (this.recordsFound) {
+        this.args.updateFogEffect(this.qualityIndex);
```

Comment:
I think this is not really part of loading data from the API and should maybe be in `loadMeasurements` instead?